File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans="";
        int count=INT_MAX;
        for(int i=0;i<strs.size();i++){
      int s=strs[i].size();
      count=min(count,s);
        }
        for(int i=0;i<count;i++){
            char ch=strs[0][i];
            for(int j=1;j<strs.size();j++){
                if(ch!=strs[j][i]){
                    return ans;
                } 
            }
               ans=ans+ch;
        }
        return ans;

    }
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        ans = ""
     
        count = float('inf')
        for s in strs:
            count = min(count, len(s))
     
        for i in range(count):
            ch = strs[0][i]
            for j in range(1, len(strs)):
                if strs[j][i] != ch:
                    return ans
            ans += ch
        
        return ans
